deploy/awstools/awstools.py

```diff
@@ -344,19 +344,34 @@ def auto_create_bucket(userbucketname):
             rootLogger.critical(repr(exc))
             assert False
 
-def subscribe_to_firesim_topic(email):
-    """ Subscribe a user to their FireSim SNS topic for notifications. """
+def get_snsname_arn():
+    """ If the Topic doesn't exist create it, send catch exceptions while creating. Or if it exists get arn """
     client = boto3.client('sns')
 
     aws_resource_names_dict = aws_resource_names()
     snsname = aws_resource_names_dict['snsname']
 
-    # this will either create the topic, if it doesn't exist, or just get the arn
-    response = client.create_topic(
-        Name=snsname
-    )
-    arn = response['TopicArn']
+    response = None
+    try: # this will either create the topic, if it doesn't exist, or just get the arn
+        response = client.create_topic(
+            Name=snsname
+        )
+    except client.exceptions.ClientError as err:
+        if 'AuthorizationError' in repr(err): 
+            rootLogger.critical("You don't have permissions to perform \"Topic Creation \". Required to send you email notifications. Please contact your IT administrator")
+
+        else:
+            rootLogger.critical("Unknown exception is encountered while trying to perform \"Topic Creation\"")
+
+        rootLogger.critical(err)
+        assert False
```

Comment:
```suggestion
    except client.exceptions.ClientError as err:
        if 'AuthorizationError' in repr(err): 
            rootLogger.warning("Will not be able to send you email notifications.  You do not have AWS IAM permission  \"sns:CreateTopic\". Please contact your administrator")
            rootLogger.debug(err)

        else:
            rootLogger.critical("Unknown exception is encountered while trying to perform \"Topic Creation\". Please notify the FireSim team.")
            rootLogger.critical(err)
            assert False
```

@rahulb10 I realized that somehow we've steered away from the original intent here.  It definitely shouldn't be a fatal error when we're unable to create the topic and we know it was for AuthError.  That would prevent people from being able to build without the notifications.  It's debatable whether the unknown exception should `assert False` but maybe it should to encourage folks to let us know about other failures we could handle more helpfully.

I'm realizing that we probably need to update the docs to describe this and also update the code used in `firesim managerinit` where it tries to create the topic and subscribe to it.  Subscription is a separate permission and could fail differently.

Reply:
If we change the AuthError to be non-fatal (like I think we should), we'll also need to update the test I added becuase it expects the AuthError to be fatal.